packages/store-sync/src/dozer/decodeRecords.ts

```diff
@@ -0,0 +1,41 @@
+import { Schema } from "@latticexyz/config";
+import { getSchemaPrimitives } from "@latticexyz/protocol-parser/internal";
+import { decodeField } from "./decodeField";
+
+type DozerQueryHeader = string[];
+type DozerQueryRecord = (string | boolean | string[])[];
+
+// First item in the result is the header
+export type DozerQueryResult = [DozerQueryHeader, ...DozerQueryRecord[]];
+
+export type DecodeDozerRecordsArgs = {
+  schema: Schema;
+  records: DozerQueryResult;
+};
+
+/**
+ * Trim the header row from the query result
+ */
+function trimHeader(result: DozerQueryResult): DozerQueryRecord[] {
+  return result.slice(1);
+}
+
+export type DecodeDozerRecordsResult<schema extends Schema = Schema> = getSchemaPrimitives<schema>[];
```

Comment:
should we update all the type names without "Dozer"?

Reply:
ahh yes missed that one, good catch